internal/inventory/azurefetcher/fetcher_resource_graph.go

```diff
@@ -107,6 +107,7 @@ func (f *resourceGraphFetcher) fetch(ctx context.Context, resourceName, serviceN
 					Name: vmProperties.Extended.InstanceView.ComputerName,
 					Type: vmProperties.HardwareProfile.VmSize,
 				}
+				asset.Host.Entity = &asset.Entity
```

Comment:
Not necessary related to your changes, but I wonder why we don't use the `inventory.WithHost` utility here.

Reply:
Not with linting or something like that, maybe unit tests can help enforce it, but I'd prefer the tests to be focus on a single class and not on the utilities.